From aa1cae95a4e54434c53dfbca2912545d71413e95 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 8 Dec 2023 15:42:59 +0100
Subject: [PATCH 0546/1695] drivers: stm32_remoteproc: add support of the
 stm32mp25 Cortex-M33

Add the support of the stm32mp25 Cortex-M33.
This first patch only enable the support of the "st,stm32mp1-m33-tee"
compatible to allow in a second step to add specific code associated
to this remote processor management ( trust zone, boot address, ....).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9eecbe74f23acb3d11a5f7f23fba1df81d7b9ef6 drivers: stm32_remoteproc: add support of the stm32mp25 Cortex-M33]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I48638bfbcf34eebf8276f27cf5ead66cdb3d2db9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/348400
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/drivers/remoteproc/stm32_remoteproc.c | 10 +++++++++-
 core/include/drivers/stm32_remoteproc.h    |  3 ++-
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index e4cd6c8b1..65099832b 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -370,7 +370,11 @@ err:
 }
 
 static const struct stm32_rproc_compat_data stm32_rproc_m4_compat = {
-	.rproc_id = STM32_M4_RPROC_ID,
+	.rproc_id = STM32MP1_M4_RPROC_ID,
+};
+
+static const struct stm32_rproc_compat_data stm32_rproc_m33_compat = {
+	.rproc_id = STM32MP2_M33_RPROC_ID,
 };
 
 static const struct dt_device_match stm32_rproc_match_table[] = {
@@ -378,6 +382,10 @@ static const struct dt_device_match stm32_rproc_match_table[] = {
 		.compatible = "st,stm32mp1-m4-tee",
 		.compat_data = &stm32_rproc_m4_compat,
 	},
+	{
+		.compatible = "st,stm32mp2-m33-tee",
+		.compat_data = &stm32_rproc_m33_compat,
+	},
 	{ }
 };
 
diff --git a/core/include/drivers/stm32_remoteproc.h b/core/include/drivers/stm32_remoteproc.h
index 8cb1efd86..5fd81ebc3 100644
--- a/core/include/drivers/stm32_remoteproc.h
+++ b/core/include/drivers/stm32_remoteproc.h
@@ -11,7 +11,8 @@
 #include <types_ext.h>
 
 /* IDs of the supported remote processors*/
-#define STM32_M4_RPROC_ID 0
+#define STM32MP1_M4_RPROC_ID 0
+#define STM32MP2_M33_RPROC_ID 1
 
 /*
  * stm32_rproc_get() - get the rproc handle associated to a remote processor ID
-- 
2.47.2

